strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/ProxyRepositoryProvider.java

```diff
@@ -206,4 +207,15 @@ public void close()
         }
     }
 
+    @Override
+    public List<Path> search(String storageId,
+                             String repositoryId,
+                             Map<String, String> coordinates,
+                             int skip,
+                             int limit,
+                             String orderBy)
+    {
+        return null;
```

Comment:
@carlspring have you any thoughts how to call `NugetRepositoryFeatures` here? :)

Reply:
we can do something like this:
```
RepositoryFeatures.processEvent(RepositoryEvent e)
```

and implement it depends on event type and ConreteRepositoryFeatures

Reply:
Why do you actually need the `NugetRepositoryFeatures` here? Are you trying to get a hold of the `NugetSearchProvider`, or what? You could inject the `SearchProviderRegistry` here and look it up, if you need to.

I'm not sure I get what your idea here is regarding the implementation of an even listener.

Perhaps I'm not fully understanding your question, or problem, so please feel free to further elaborate. :)